### ISSUE TITLE
Unknown error

### ISSUE THREAD

**Issue by greenygh0st** [hidapple/graylog-plugin-teams#11]
I am receiving the following message when trying to run a test notification for the plugin:

`Unknown errorResponse, please check your Graylog server logs.`

But when I check the Graylog server logs under `/var/log/graylog-server` I find no error messages.

![image](https://user-images.githubusercontent.com/1551020/69664163-69b1af00-104d-11ea-82a5-959852cec5ab.png)

"..." are taking the place of the actual secret strings.

Any ideas? I realize this is not a lot to go on but it is all I have. :)

**Comment by jthunt** [comment]
I am still using the legacy plugin right now..   I tried to upgrade to the 2.0 version and got the same error.   I am using Graylog docker version 3.1.3-1 (also tried most versions between 3.1.0 and 3.1.3).

When you click Execute Test Notification, you see the error "Error: Unknown error, please check your Graylog server logs."

Nothing in the logs but I did notice a 500 error in the post request.

> Internal Server Error
> Reason
> java.lang.NoSuchMethodError: org.graylog.events.notifications.EventNotificationContext.jobTrigger()Lorg/graylog/scheduler/JobTriggerDto;
> 
> Stacktrace
> org.glassfish.jersey.server.ContainerException: java.lang.NoSuchMethodError: org.graylog.events.notifications.EventNotificationContext.jobTrigger()Lorg/graylog/scheduler/JobTriggerDto;
> 	at org.glassfish.jersey.grizzly2.httpserver.GrizzlyHttpContainer$ResponseWriter.rethrow(GrizzlyHttpContainer.java:324)
> 	at org.glassfish.jersey.grizzly2.httpserver.GrizzlyHttpContainer$ResponseWriter.failure(GrizzlyHttpContainer.java:306)
> 	at org.glassfish.jersey.server.ServerRuntime$Responder.process(ServerRuntime.java:509)
> 	at org.glassfish.jersey.server.ServerRuntime$2.run(ServerRuntime.java:334)
> 	at org.glassfish.jersey.internal.Errors$1.call(Errors.java:271)
> 	at org.glassfish.jersey.internal.Errors$1.call(Errors.java:267)
> 	at org.glassfish.jersey.internal.Errors.process(Errors.java:315)
> 	at org.glassfish.jersey.internal.Errors.process(Errors.java:297)
> 	at org.glassfish.jersey.internal.Errors.process(Errors.java:267)
> 	at org.glassfish.jersey.process.internal.RequestScope.runInScope(RequestScope.java:317)
> 	at org.glassfish.jersey.server.ServerRuntime.process(ServerRuntime.java:305)
> 	at org.glassfish.jersey.server.ApplicationHandler.handle(ApplicationHandler.java:1154)
> 	at org.glassfish.jersey.grizzly2.httpserver.GrizzlyHttpContainer.service(GrizzlyHttpContainer.java:384)
> 	at org.glassfish.grizzly.http.server.HttpHandler$1.run(HttpHandler.java:224)
> 	at com.codahale.metrics.InstrumentedExecutorService$InstrumentedRunnable.run(InstrumentedExecutorService.java:181)
> 	at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1149)
> 	at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:624)
> 	at java.lang.Thread.run(Thread.java:748)
> Caused by: java.lang.NoSuchMethodError: org.graylog.events.notifications.EventNotificationContext.jobTrigger()Lorg/graylog/scheduler/JobTriggerDto;
> 	at org.graylog.plugins.teams.event.notifications.TeamsEventNotification.getModel(TeamsEventNotification.java:62)
> 	at org.graylog.plugins.teams.event.notifications.TeamsEventNotification.execute(TeamsEventNotification.java:47)
> 	at org.graylog.events.notifications.NotificationResourceHandler.test(NotificationResourceHandler.java:181)
> 	at org.graylog.events.rest.EventNotificationsResource.test(EventNotificationsResource.java:196)
> 	at sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
> 	at sun.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
> 	at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
> 	at java.lang.reflect.Method.invoke(Method.java:498)
> 	at org.glassfish.jersey.server.model.internal.ResourceMethodInvocationHandlerFactory$1.invoke(ResourceMethodInvocationHandlerFactory.java:81)
> 	at org.glassfish.jersey.server.model.internal.AbstractJavaResourceMethodDispatcher$1.run(AbstractJavaResourceMethodDispatcher.java:144)
> 	at org.glassfish.jersey.server.model.internal.AbstractJavaResourceMethodDispatcher.invoke(AbstractJavaResourceMethodDispatcher.java:161)
> 	at org.glassfish.jersey.server.model.internal.JavaResourceMethodDispatcherProvider$ResponseOutInvoker.doDispatch(JavaResourceMethodDispatcherProvider.java:160)
> 	at org.glassfish.jersey.server.model.internal.AbstractJavaResourceMethodDispatcher.dispatch(AbstractJavaResourceMethodDispatcher.java:99)
> 	at org.glassfish.jersey.server.model.ResourceMethodInvoker.invoke(ResourceMethodInvoker.java:389)
> 	at org.glassfish.jersey.server.model.ResourceMethodInvoker.apply(ResourceMethodInvoker.java:347)
> 	at org.glassfish.jersey.server.model.ResourceMethodInvoker.apply(ResourceMethodInvoker.java:102)
> 	at org.glassfish.jersey.server.ServerRuntime$2.run(ServerRuntime.java:326)
> 	... 14 more

**Comment by hidapple** [comment]
Thanks for reporting the problem.
It seems that test notification button was implemented starting in Graylog 3.1.3
https://docs.graylog.org/en/3.1/pages/changelog.html#graylog-3-1-3

I will check how to support that function by teams plugin!
Thank you!

**Comment by greenygh0st** [comment]
Awesome! Thanks for looking into it!

**Comment by jthunt** [comment]
@hidapple Just an FYI..   It doesn't seem to be just a sending test notification issue.  I configured the notification and it doesn't actually trigger.  If I look at Graylog's logs, I see a similar error.

> graylog_graylog.1.2qpenlkst5kr@jamesvm    | 2019-12-02 14:53:54,263 ERROR: org.graylog.scheduler.worker.JobWorkerPool - Unhandled exception
> graylog_graylog.1.2qpenlkst5kr@jamesvm    | java.lang.NoSuchMethodError: org.graylog.events.notifications.EventNotificationContext.jobTrigger()Lorg/graylog/scheduler/JobTriggerDto;
> graylog_graylog.1.2qpenlkst5kr@jamesvm    |     at org.graylog.plugins.teams.event.notifications.TeamsEventNotification.getModel(TeamsEventNotification.java:62) ~[?:?]
> graylog_graylog.1.2qpenlkst5kr@jamesvm    |     at org.graylog.plugins.teams.event.notifications.TeamsEventNotification.execute(TeamsEventNotification.java:47) ~[?:?]
> graylog_graylog.1.2qpenlkst5kr@jamesvm    |     at org.graylog.events.notifications.EventNotificationExecutionJob.execute(EventNotificationExecutionJob.java:135) ~[graylog.jar:?]
> graylog_graylog.1.2qpenlkst5kr@jamesvm    |     at org.graylog.scheduler.JobExecutionEngine.executeJob(JobExecutionEngine.java:166) ~[graylog.jar:?]
> graylog_graylog.1.2qpenlkst5kr@jamesvm    |     at org.graylog.scheduler.JobExecutionEngine.lambda$handleTrigger$2(JobExecutionEngine.java:144) ~[graylog.jar:?]
> graylog_graylog.1.2qpenlkst5kr@jamesvm    |     at com.codahale.metrics.Timer.time(Timer.java:137) ~[graylog.jar:?]
> graylog_graylog.1.2qpenlkst5kr@jamesvm    |     at org.graylog.scheduler.JobExecutionEngine.handleTrigger(JobExecutionEngine.java:144) ~[graylog.jar:?]
> graylog_graylog.1.2qpenlkst5kr@jamesvm    |     at org.graylog.scheduler.JobExecutionEngine.lambda$execute$0(JobExecutionEngine.java:119) ~[graylog.jar:?]
> graylog_graylog.1.2qpenlkst5kr@jamesvm    |     at org.graylog.scheduler.worker.JobWorkerPool.lambda$execute$0(JobWorkerPool.java:110) ~[graylog.jar:?]
> graylog_graylog.1.2qpenlkst5kr@jamesvm    |     at com.codahale.metrics.InstrumentedExecutorService$InstrumentedRunnable.run(InstrumentedExecutorService.java:181) ~[graylog.jar:?]
> graylog_graylog.1.2qpenlkst5kr@jamesvm    |     at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1149) ~[?:1.8.0_232]
> graylog_graylog.1.2qpenlkst5kr@jamesvm    |     at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:624) ~[?:1.8.0_232]
> graylog_graylog.1.2qpenlkst5kr@jamesvm    |     at com.codahale.metrics.InstrumentedThreadFactory$InstrumentedRunnable.run(InstrumentedThreadFactory.java:66) ~[graylog.jar:?]
> graylog_graylog.1.2qpenlkst5kr@jamesvm    |     at java.lang.Thread.run(Thread.java:748) [?:1.8.0_232]

**Comment by JakubJachym** [comment]
Hey @hidapple!
I can confirm version 1.1.1 is working in a Legacy notification mode in Graylog 3.1.3. Even the Test Notification button works.

However version 2.0 doesn't work at all, throwing an exception into log:
```
ERROR [JobWorkerPool] Unhandled exception
java.lang.NoSuchMethodError: org.graylog.events.notifications.EventNotificationContext.jobTrigger()Lorg/graylog/scheduler/JobTriggerDto;
        at org.graylog.plugins.teams.event.notifications.TeamsEventNotification.getModel(TeamsEventNotification.java:62) ~[?:?]
        at org.graylog.plugins.teams.event.notifications.TeamsEventNotification.execute(TeamsEventNotification.java:47) ~[?:?]
        at org.graylog.events.notifications.EventNotificationExecutionJob.execute(EventNotificationExecutionJob.java:135) ~[graylog.jar:?]
        at org.graylog.scheduler.JobExecutionEngine.executeJob(JobExecutionEngine.java:166) ~[graylog.jar:?]
        at org.graylog.scheduler.JobExecutionEngine.lambda$handleTrigger$2(JobExecutionEngine.java:144) ~[graylog.jar:?]
        at com.codahale.metrics.Timer.time(Timer.java:137) ~[graylog.jar:?]
        at org.graylog.scheduler.JobExecutionEngine.handleTrigger(JobExecutionEngine.java:144) ~[graylog.jar:?]
        at org.graylog.scheduler.JobExecutionEngine.lambda$execute$0(JobExecutionEngine.java:119) ~[graylog.jar:?]
        at org.graylog.scheduler.worker.JobWorkerPool.lambda$execute$0(JobWorkerPool.java:110) ~[graylog.jar:?]
        at com.codahale.metrics.InstrumentedExecutorService$InstrumentedRunnable.run(InstrumentedExecutorService.java:181) ~[graylog.jar:?]
        at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1149) ~[?:1.8.0_201]
        at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:624) ~[?:1.8.0_201]
        at com.codahale.metrics.InstrumentedThreadFactory$InstrumentedRunnable.run(InstrumentedThreadFactory.java:66) ~[graylog.jar:?]
        at java.lang.Thread.run(Thread.java:748) [?:1.8.0_201]
```

I hope it helps to debug it.

**Comment by hidapple** [comment]
Thank you guys for giving me helpful information!
I have just published version 2.0.1 which works with Graylog v3.1.3's test notification function.

**Comment by JakubJachym** [comment]
Thanks! I can confirm it works and all (not just test) notifications are sent to MS Teams as expected.

Great work! Thanks again!
